src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IfError.cs

```diff
@@ -96,77 +96,80 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
-                //   IfError(v1, f1, v2, f2, v3) --> possible values to be returned are f1, f2 or v3
-                var typeCanBeReturned = (i % 2) == 1 || i == ((count % 2) == 0 ? (count - 2) : (count - 1));
-
-                if (typeCanBeReturned)
+                if (!typeSuper.IsError)
                 {
-                    // Let's check if it matches the other types that can be returned
-                    var typeSuper = DType.Supertype(type, typeArg);
-
-                    if (!typeSuper.IsError)
+                    // If preferred type was error or null (due to Blank or Error) assign new type in hierarchy
+                    if (type.IsError || type == DType.ObjNull)
                     {
                         type = typeSuper;
                     }
-                    else if (type.Kind == DKind.Unknown)
+                }
+                else
+                {
+                    if (typeArg.IsVoid)
+                    {
+                        type = DType.Void;
+                    }
+                    else if (typeArg.IsError)
                     {
-                        // One of the args is also of unknown type, so we can't resolve the type of IfError
-                        type = typeSuper;
+                        errors.EnsureError(args[i], TexlStrings.ErrTypeError);
                         fArgsValid = false;
                     }
-                    else if (!type.IsError)
+                    else if (!type.IsError && typeArg.CoercesTo(type))
                     {
-                        // Types don't resolve normally, coercion needed
-                        if (typeArg.CoercesTo(type))
-                        {
-                            CollectionUtils.Add(ref nodeToCoercedTypeMap, nodeArg, type);
-                        }
-                        else if (!isBehavior || !IsArgTypeInconsequential(nodeArg))
-                        {
-                            errors.EnsureError(
-                                DocumentErrorSeverity.Severe,
-                                nodeArg,
-                                TexlStrings.ErrBadType_ExpectedType_ProvidedType,
-                                type.GetKindString(),
-                                typeArg.GetKindString());
-                            fArgsValid = false;
-                        }
+                        CollectionUtils.Add(ref nodeToCoercedTypeMap, nodeArg, type);
                     }
-                    else if (typeArg.Kind != DKind.Unknown)
+                    else
                     {
-                        type = typeArg;
-                        fArgsValid = false;
+                        type = DType.Void;
                     }
                 }
 
-                // If there are an odd number of args, the last arg also participates.
-                i += 2;
-                if (i == count)
+                if ((count % 2) != 0 && i == count - 1)
                 {
```

Comment:
Since you already handled the odd argument case (preferredTypeIndex on line 126-127), do you still need this?

Reply:
Yes, because I need to start at the end of arg list (for both even and odd).
e.g. even arg case `IfError(v1, f1, v2, f2)` 
preferred return type is v2. But f2 can also be returned and need to type check it.

When it comes to odd arg
e.g. odd arg case `IfError(v1, f1, v2, f2, v3)`
preferred return type is v3. but f1 or f2 could also possibly be returned. so this line adjusts loop to scan f1, f2 and typecheck them.